src/core/sock/sockinfo_tcp.cpp

```diff
@@ -4556,6 +4556,42 @@ int sockinfo_tcp::tcp_setsockopt(int __level, int __optname, __const void *__opt
             ret = -1;
             errno = EINVAL;
             break;
+        case SO_XLIO_ISOLATE:
+            pass_to_os_cond = false;
+            if (__optlen == sizeof(int)) {
+                if (!m_p_connected_dst_entry && m_rx_ring_map.empty()) {
+                    int tempval = *reinterpret_cast<const int *>(__optval);
+                    bool ring_isolated =
+                        m_ring_alloc_log_rx.get_ring_alloc_logic() == RING_LOGIC_ISOLATE &&
+                        m_ring_alloc_log_tx.get_ring_alloc_logic() == RING_LOGIC_ISOLATE;
+
+                    if (tempval == SO_XLIO_ISOLATE_DEFAULT && !ring_isolated) {
```

Comment:
suggested condition will break (and return success) in case of SO_XLIO_ISOLATE_DEFAULT **and** ring_isolated. But we must return EINVAL error in this case.

Since this is a control flow and proper application will unlikely call few times SO_XLIO_ISOLATE_SAFE in a row, lets keep logic as is and execute the branch with no useful effect. I don't see how to return success in this flow without complicating the code.